### PR TITLE
fix: replace firefly mute system

### PR DESCRIPTION
**Pull request by dudu0506**
https://mask.atlassian.net/browse/FW-1967

# Pull Request

## Related Issue

Closes FW-0000

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/app/api/twitter/users/route.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                'description',
+                'username',
+                'name',
+                'profile_image_url',
+                'public_metrics',
+                'connection_status',
+            ],
+        });
+
+        if (errors && errors.length > 0) {
```
Comment (guanbinrui) [inline src/app/api/twitter/users/route.ts:37]: Build into getTwitterErrorMessage

File: src/services/getProfilesByIds.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            return await NeynarSocialMediaProvider.getProfilesByIds(ids);
+        case SourceInURL.Lens:
+            return await LensSocialMediaProvider.getProfilesByIds(ids);
+        case SourceInURL.Twitter:
+            return await TwitterSocialMediaProvider.getProfilesByIds(ids);
+        case SourceInURL.Firefly:
+        case SourceInURL.Article:
+        case SourceInURL.Wallet:
+        case SourceInURL.NFTs:
+            throw new UnreachableError('Not implemented', source);
```
Comment (guanbinrui) [inline src/services/getProfilesByIds.ts:19]: NotImplementedError

File: src/helpers/getTwitterErrorMessage.ts [hunk 3]
```diff
@@ -15,19 +17,26 @@ interface TwitterError {
...
-    if (!(error instanceof Error)) return 'Unknown error.';
+export function getTwitterErrorMessage(error: unknown): { status?: number; message: string } {
+    // inline error handling
+    if (Array.isArray(error)) {
+        const forbiddenError = error.find(({ title }) => title === 'Forbidden');
+        if (forbiddenError) return { status: StatusCodes.FORBIDDEN, message: forbiddenError.detail };
+        return { message: error.map(({ title, detail }) => `${title}: ${detail}`).join('\n') };
+    }
+
+    if (!(error instanceof Error)) return { message: 'Unknown error.' };
```
Comment (guanbinrui) [inline src/helpers/getTwitterErrorMessage.ts:28]: Below code will unreachable.

File: src/helpers/runInSafe.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { IS_PRODUCTION } from '@/constants/index.js';
+
+export async function runInSafe<T>(fn: () => Promise<T>, noThrow = true) {
+    try {
+        await fn();
+    } catch (error) {
+        if (IS_PRODUCTION) return;
```
Comment (guanbinrui) [inline src/helpers/runInSafe.ts:7]: We should throw in production.

File: src/helpers/getTwitterErrorMessage.ts (hunk 3 above)
Comment (guanbinrui) [inline src/helpers/getTwitterErrorMessage.ts:28]:
```suggestion
    if (!(error instanceof Error)) return { message: 'Unknown error.' };

    // inline error handling
    if (Array.isArray(error)) {
        const forbiddenError = error.find(({ title }) => title === 'Forbidden');
        if (forbiddenError) return { status: StatusCodes.FORBIDDEN, message: forbiddenError.detail };
        return { message: error.map(({ title, detail }) => `${title}: ${detail}`).join('\n') };
    }
```

File: src/helpers/runInSafe.ts (hunk 4 above)
Comment (dudu0506) [reply src/helpers/runInSafe.ts:7]: This method aims to run safe with no error, so it wont affect other logic

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**